client.cpp

```diff
@@ -878,6 +878,130 @@ bool verify_client::finished(void)
 
 ///////////////////////////////////////////////////////////////////////////
 
+crc_verify_client::verify_request::verify_request(request_type type,
+                                              unsigned int size,
+                                              struct timeval* sent_time,
+                                              unsigned int keys,
+                                              const char *key,
+                                              unsigned int key_len) :
+        client::request(type, size, sent_time, keys),
+        m_key(NULL), m_key_len(0)
+{
+    m_key_len = key_len;
+    m_key = (char *)malloc(key_len);
+    memcpy(m_key, key, m_key_len);
+}
+
+crc_verify_client::verify_request::~verify_request(void)
+{
+    if (m_key != NULL) {
+        free((void *) m_key);
+        m_key = NULL;
+    }
+}
+
+crc_verify_client::crc_verify_client(struct event_base *event_base,
+                             benchmark_config *config,
+                             abstract_protocol *protocol,
+                             object_generator *obj_gen) : client(event_base, config, protocol, obj_gen),
+                                                          m_finished(false), m_verified_keys(0), m_errors(0)
+{
+    m_protocol->set_keep_value(true);
+    m_obj_gen->set_key_range(config->key_minimum, config->key_maximum);
+}
+
+unsigned long long int crc_verify_client::get_verified_keys(void)
+{
+    return m_verified_keys;
+}
+
+unsigned long long int crc_verify_client::get_errors(void)
+{
+    return m_errors;
+}
+
+void crc_verify_client::create_request(struct timeval timestamp)
+{
+    // Prepare a GET request that will be compared against a previous
```

Comment:
how is the previous SET command is determined? do you send commands in pairs of (SET, GET)?

Reply:
No, I currently rely on the key generation order and fact the it is simply sequential. When initializing the verify client I reset the keys of the object generator, that ensures that we'll start from the beginning of the key range and continue to generate keys sequentially till the end of the range.

Reply:
@yael-lb  i dont understand how does it work, will it work with compression data ?